opentelemetry/proto/resource/v1/resource.proto

```diff
@@ -42,4 +42,22 @@ message Resource {
   //
   // Status: [Development]
   repeated opentelemetry.proto.common.v1.EntityRef entity_refs = 3;
+
+  // Set of attributes that describe the resource.
+  // Key and Value combination MUST be unique.
+  // attributes MUST NOT be used if dict_attributes is used.
+  //
+  // Note: This is currently used exclusively in the Profiling signal.
+  //
+  // Status: [Development]
+  repeated RefKeyValue dict_attributes = 4;
```

Comment:
I would put the dictionary in the "ResourceLogs/ResourceSpans/ResourceMetrics" and be shared between all attributes coming from a resource?

happy to hear others what they think

Reply:
>If we scope the dictionary to each Resource or inside ResourceLogs/ResourceSpans/ResourceMetrics, it can’t de-duplicate across separate Resource instances once we move to per-process resources. That defeats the goal and wouldn’t prevent the payload growth we’re trying to avoid.

Do you think you can run the benchmarks with both approaches (what you suggest and what Bogdan suggests) so that we can see the difference?